### PR TITLE
Store result in store instead of result hash

### LINKED ISSUE #1881: Store result in store instead of result hash
It makes other can query result from rest-endpoint

Tasks
- [ ] Save result (OBI encode of request + response packet) to storage at endblock logic
- [ ] Fix get proof endpoint (no need to find from end block event)
- [ ] Update solidity code and test

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed #1881

### CODE REVIEW

File: bridges/evm/contracts/Bridge.sol [hunk 1]
```diff
@@ -130,7 +130,7 @@ contract Bridge is IBridge, Ownable {
         vars.encodedVarint = Utils.encodeVarintSigned(_version);
         vars.dataHash = sha256(
             abi.encodePacked(
```
Comment (sorawit) [inline bridges/evm/contracts/Bridge.sol]: I think we can remove `abi.encodePacked`

File: chain/x/oracle/keeper/request.go [hunk 2]
```diff
@@ -88,8 +88,8 @@ func (k Keeper) SaveResult(ctx sdk.Context, id types.RequestID, status types.Res
 		r.ClientID, id, k.GetReportCount(ctx, id), r.RequestTime,
 		ctx.BlockTime().Unix(), status, result,
 	)
-	resultHash := types.CalculateResultHash(req, res)
-	k.SetResult(ctx, id, resultHash)
+	encodedResult := types.CalculateEncodedResult(req, res)
```
Comment (sorawit) [inline chain/x/oracle/keeper/request.go]: Let's make this one liner.

File: chain/x/oracle/types/packets.go [hunk 3]
```diff
@@ -17,9 +16,9 @@ func (p OracleResponsePacketData) GetBytes() []byte {
 	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(p))
 }
 
-// CalculateResultHash returns hash of append hash of request and response packet.
-func CalculateResultHash(req OracleRequestPacketData, res OracleResponsePacketData) []byte {
-	reqPacketHash := tmhash.Sum(obi.MustEncode(req))
-	resPacketHash := tmhash.Sum(obi.MustEncode(res))
-	return tmhash.Sum(append(reqPacketHash, resPacketHash...))
+// CalculateEncodedResult returns append obi encode of request and response packet.
+func CalculateEncodedResult(req OracleRequestPacketData, res OracleResponsePacketData) []byte {
```
Comment (sorawit) [inline chain/x/oracle/types/packets.go:20]: Can make this one line!

File: chain/x/oracle/types/packets.go [hunk 4]
```diff
@@ -17,9 +16,9 @@ func (p OracleResponsePacketData) GetBytes() []byte {
 	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(p))
 }
 
-// CalculateResultHash returns hash of append hash of request and response packet.
-func CalculateResultHash(req OracleRequestPacketData, res OracleResponsePacketData) []byte {
-	reqPacketHash := tmhash.Sum(obi.MustEncode(req))
-	resPacketHash := tmhash.Sum(obi.MustEncode(res))
-	return tmhash.Sum(append(reqPacketHash, resPacketHash...))
+// CalculateEncodedResult returns append obi encode of request and response packet.
```
Comment (sorawit) [inline chain/x/oracle/types/packets.go]: obi encoded of ... packets.

File: chain/x/oracle/types/events.go [hunk 5]
```diff
@@ -30,5 +30,5 @@ const (
 	AttributeKeyResolveTime    = "resolve_time"
 	AttributeKeyResolveStatus  = "resolve_status"
 	AttributeKeyResult         = "result"
-	AttributeKeyResultHash     = "result_hash"
+	AttributeKeyEncodedResult  = "encoded_result"
```
Comment (sorawit) [inline chain/x/oracle/types/events.go]: Let's remove this event.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/oracle/types/packets_test.go [hunk 6]
```diff
@@ -57,11 +57,11 @@ func TestCalculateResultHash(t *testing.T) {
 		ResolveStatus: ResolveStatus(1),
 		Result:        mustDecodeString("4bb10e0000000000"),
 	}
-	expectedResultHash := mustDecodeString("29bcc52d59b39c61a9616365dcd39dbff8d1aebc88a6a7e2b53dff67841dbc06")
-	require.Equal(t, expectedResultHash, CalculateResultHash(req, res))
+	expectedEncodedResult := []byte{0x0, 0x0, 0x0, 0x4, 0x62, 0x65, 0x65, 0x62, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0xf, 0x3, 0x0, 0x0, 0x0, 0x42, 0x54, 0x43, 0x64, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x4, 0x62, 0x65, 0x65, 0x62, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x5e, 0xbe, 0x61, 0x2c, 0x0, 0x0, 0x0, 0x0, 0x5e, 0xbe, 0x61, 0x2e, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x8, 0x4b, 0xb1, 0xe, 0x0, 0x0, 0x0, 0x0, 0x0}
```
Comment (sorawit) [inline chain/x/oracle/types/packets_test.go]: Please use `mustDecodeString`

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**